keras_cv/applications/object_detection/retina_net/retina_net.py

```diff
@@ -0,0 +1,299 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import tensorflow as tf
+from tensorflow import keras
+
+from keras_cv import bounding_box
+from keras_cv.applications.object_detection.retina_net.__internal__ import (
+    layers as layers_lib,
+)
+from keras_cv.applications.object_detection.retina_net.__internal__ import (
+    utils as utils_lib,
+)
+
+
+class RetinaNet(keras.Model):
+    """A Keras model implementing the RetinaNet architecture.
+
+    RetinaNet implements the RetinaNet architecture for object detection.  The class
+    requires a number of classes, bounding_box format and a backbone.  Optionally, a
+    custom label encoder, feature pyramid network, and prediction decoder may all be
+    provided.  Please see {LINK} for an example of how to implement a custom version of
+    each of these components.
+
+    Usage:
+    ```
+    retina_net = keras_cv.applications.RetinaNet(
+        num_classes=20,
+        bounding_box_format="xywh",
+        backbone="resnet50",
+        backbone_weights="imagenet",
+        include_rescaling=True,
+    )
+    ```
+
+    Args:
+        num_classes:
```

Comment:
There are 5 args missing docs -- I guess that's intentional for now? Just wanted to check

Reply:
Done